mailbox/api/src/main/java/org/apache/james/mailbox/Event.java

```diff
@@ -24,13 +24,9 @@
 
 public interface Event {
 
-    MailboxSession getSession();
-
-    default User getUser() {
-        return getSession().getUser().getCoreUser();
-    }
+    User getUser();
 
     default Optional<MailboxSession.SessionId> getSessionId() {
```

Comment:
do you need a default implementation for that ?

Reply:
Having default method helps me avoid to override this when I dont need it. E.g the event `QuotaUsageUpdatedEvent` doesn't carry a `sessionId`. What did you doubt?

Reply:
1/ why do you need a getSessionId method when you don't have one. It looks like unneeded code factorization
2/ are we sure a QuotaUpdate doesn't need a session id at the business level ? What makes it different from other events in that respect ?

Reply:
I realized Only MailboxEvent need sessionId :( Then yeah, the Optional SessionId is a waste. updated in https://github.com/linagora/james-project/pull/1980/commits/007943646873a65b432023586699dd08a70636df

I checked other Event Children in code (MessageMoveEvent, QuotaUsageUpdatedEvent). They dont need sessionId.